### ISSUE TITLE
copy module always returns "changed: false" when recursively copying a directory

### ISSUE THREAD

**Issue by TrLucas** [ansible/ansible#53958]
##### SUMMARY
When attempting to recursively copy a directory and registering the result, the result's `changed` parameter yields `false`, even though directories where newly created / files newly copied.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

modules/files/copy.py

##### ANSIBLE VERSION

```paste below
ansible 2.7.9
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/tlucas/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]
```

##### OS / ENVIRONMENT

Both the controlling and the remote machine run on Debian 9.8

##### STEPS TO REPRODUCE

* Create a directory `test_copy` along with some nested directories / files in your current working directory (I executed this with having a file `actual.file` located in `test_copy/foo/bar`, which also appears on the remote host in the correct location)
* Run the below playbook (located also in your current working directory, `YOUR_REMOTE_HOST` replaced with an actual existing host you have access to)

```yaml
- name:
    "copy-recursively"

  hosts:
    "YOUR_REMOTE_HOST"

  tasks:
    - name:
        "recursively copy the local folder 'test_copy' to the remote machine"
      copy:
        src: "test_copy"
        dest: "/opt/"
      register: "copied"

    - debug:
        msg: "{{ copied }}"
```

##### EXPECTED RESULTS

`copied` should have `changed: true`, which then a) marks the task as actually being executed and b) enables subsequent processing of when the directories content has changed.

##### ACTUAL RESULTS

```paste below
PLAY [copy-recursively] ******************************************************************************

TASK [Gathering Facts] ************************************************************************************************
ok: [YOUR_REMOTE_HOST]

TASK [recursively copy the local folder 'test_copy' to the remote machine] ***********************************************
ok: [YOUR_REMOTE_HOST]

TASK [debug] **********************************************************************************************************
ok: [YOUR_REMOTE_HOST] => {
    "msg": {
        "changed": false, 
        "dest": "/opt/test_copy/foo/bar", 
        "diff": {
            "after": {
                "path": "/opt/test_copy/foo/bar"
            }, 
            "before": {
                "path": "/opt/test_copy/foo/bar"
            }
        }, 
        "failed": false, 
        "gid": 0, 
        "group": "root", 
        "mode": "0755", 
        "owner": "root", 
        "path": "/opt/test_copy/foo/bar", 
        "size": 4096, 
        "state": "directory", 
        "uid": 0
    }
}

PLAY RECAP ************************************************************************************************************
YOUR_REMOTE_HOST    : ok=3    changed=0    unreachable=0    failed=0 
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/files/copy.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/copy.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @ptux
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
I've verified this, but it seems to only happen in a very specific instance.

If there is only 1 file that is copied, the task does not report changed.  If there is more than 1 file copied, then the task correctly reports changed=True.

**Comment by elarrarte-d** [comment]
Any updates on this issue? Thanks!

**Comment by sivel** [comment]
This issue has been verified, and labeled as a P3 (which means the core team believes it should be addressed, but does not block new releases).

At this point it is waiting on someone to work on and submit a fix, which has not been done.  No one is assigned directly to work on this issue.

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/copy.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/copy.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by twolsten** [comment]
any update on this?

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/38938
